services/datamanager/builtin/capture_test.go

```diff
@@ -204,3 +198,14 @@ func TestDataCaptureEnabled(t *testing.T) {
 		})
 	}
 }
+
+func waitForCaptureFiles(captureDir string) {
+	files := getAllFileInfos(captureDir)
+	for i := 0; i < 50; i++ {
+		if len(files) > 0 && files[0].Size() > int64(emptyFileBytesSize) {
+			return
+		}
+		time.Sleep(time.Millisecond * 25)
```

Comment:
I may be missing something, but we're looping for the same amount of time here, correct? So do we actually expect  this test to be less flaky?

Reply:
No this test wasn't functionally changed (and already shouldn't be flaky - unless a GC stops the world for >5 seconds). I just updated it to use the mockClock to simulate times passage for explicitness/consistency with other tests

Reply:
I actually bumped this and the other waits up.

In the not-Enabled tests, this should only impact the failure cases. These waits are only hit in the failure case, so we're trading off decreasing rates of false negatives for failure cases taking longer. And I don't think test speed when they're failing really matters much, so that I think is a good trade off.

For the enabled tests, they're a bit more annoying, because the tests are testing "make sure X _doesn't_ happen", so we actually do wait that amount of time in any of the disabled cases. Which definitely isn't perfect, but I'm not sure of a better way to ensure that something isn't happening. And I figure the cost of a single false negative in CI is worth many, many seconds so worth being conservative here